### PR TITLE
feat: External conversation list table

### PR DESCRIPTION
**Pull request by wadedvsa**
## Description

Adds the table configuration for the external conversation list.

Related issue: [VOL-4588](https://dvsa.atlassian.net/browse/VOL-4588)

## Before submitting (or marking as "ready for review")

- [ ] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [ ] Have you performed a self-review of the code
- [ ] Have you have added tests that prove the fix or feature is effective and working
- [ ] Did you make sure to update any documentation relating to this change?

### CODE REVIEW

File: Common/src/Common/Service/Table/Formatter/ExternalConversationLink.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                break;
+            case "CLOSED":
+                $tagColor = 'govuk-tag--grey';
+                break;
+            default:
+                $tagColor = 'govuk-tag--green';
+                break;
+        }
+
+        $rows = '<a class="' . 'govuk-body govuk-link govuk-!-padding-right-1 ' . $statusCSS . '" href="%s">%s: %s</a>
```
Comment (fibble) [inline Common/src/Common/Service/Table/Formatter/ExternalConversationLink.php:55]: Not one for right now, but perhaps to raise an an Improvement.. Could we re-use anything in Common/src/Common/View/Helper/ (maybe Link.php & Status.php) here? Would mean any future changes to govuk Markup need tweaking in less places.

File: Common/src/Common/Service/Table/Formatter/ExternalConversationLinkFactory.php [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Common\Service\Table\Formatter;
+
+use Interop\Container\ContainerInterface;
+use Laminas\ServiceManager\Factory\FactoryInterface;
```
Comment (fibble) [inline Common/src/Common/Service/Table/Formatter/ExternalConversationLinkFactory.php:8]:
```suggestion
use Laminas\ServiceManager\Factory\FactoryInterface;
use Common\Service\Helper\UrlHelperService;
```

File: Common/src/Common/Service/Table/Formatter/ExternalConversationLinkFactory.php [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+{
+    /**
+     * @param  ContainerInterface $container
+     * @param  $requestedName
+     * @param  array|null         $options
+     * @return InternalLicenceConversationLink
+     */
+    public function __invoke(ContainerInterface $container, $requestedName, array $options = null)
+    {
+        $urlHelper = $container->get('Helper\Url');
```
Comment (fibble) [inline Common/src/Common/Service/Table/Formatter/ExternalConversationLinkFactory.php:20]: UrlHelperService::class here would be preferable. 

```suggestion
        $urlHelper = $container->get(UrlHelperService::class);
```

**Review by fibble — CHANGES_REQUESTED** [review]

File: Common/src/Common/Service/Table/Formatter/ExternalConversationLink.php (hunk 1 above)
Comment (wadedvsa) [reply Common/src/Common/Service/Table/Formatter/ExternalConversationLink.php:55]: 👍

**Pull request closed without merging**